src/sql/parts/jobManagement/views/jobHistory.component.ts

```diff
@@ -146,18 +171,30 @@ export class JobHistoryComponent extends Disposable implements OnInit {
 	loadHistory() {
 		const self = this;
 		let ownerUri: string = this._dashboardService.connectionManagementService.connectionInfo.ownerUri;
-		this._jobManagementService.getJobHistory(ownerUri, this.jobId).then((result) => {
+		this._jobManagementService.getJobHistory(ownerUri, this._agentViewComponent.jobId).then((result) => {
 			if (result && result.jobs) {
-				self._treeController.jobHistories = result.jobs;
-				self._agentJobCacheService.setJobHistory(self.jobId, result.jobs);
-				let jobHistoryRows = self._treeController.jobHistories.map(job => self.convertToJobHistoryRow(job));
-				self._treeDataSource.data = jobHistoryRows;
-				self._tree.setInput(new JobHistoryModel());
-				self.agentJobHistoryInfo =  self._treeController.jobHistories[0];
-				if (this.agentJobHistoryInfo) {
-					self.agentJobHistoryInfo.runDate = self.formatTime(self.agentJobHistoryInfo.runDate);
+				if (result.jobs.length > 0) {
+					self._showPreviousRuns = true;
```

Comment:
this code looks similar to a code block in `ngAfterContentChecked` (though not exactly the same).  I wonder if there is a way to refactor these two blocks in to a shared method.  Would that make sense?